src/devices/chihiro/JvsIo.cpp

```diff
@@ -97,203 +118,246 @@ int JvsIo::Jvs_Command_GetBoardId()
 	return 0;
 }
 
-int JvsIo::Jvs_Command_GetCommandFormat()
+int JvsIo::Jvs_Command_11_GetCommandFormat()
 {
-	ResponseBuffer.push_back(0x01);
+	ResponseBuffer.push_back(ReportCode::Handled);
 	ResponseBuffer.push_back(CommandFormatRevision);
 
 	return 0;
 }
 
-int JvsIo::Jvs_Command_GetJvsRevision()
+int JvsIo::Jvs_Command_12_GetJvsRevision()
 {
-	ResponseBuffer.push_back(0x01);
+	ResponseBuffer.push_back(ReportCode::Handled);
 	ResponseBuffer.push_back(JvsVersion);
 
 	return 0;
 }
 
-int JvsIo::Jvs_Command_GetCommunicationVersion()
+int JvsIo::Jvs_Command_13_GetCommunicationVersion()
 {
-	ResponseBuffer.push_back(0x01);
+	ResponseBuffer.push_back(ReportCode::Handled);
 	ResponseBuffer.push_back(CommunicationVersion);
 
 	return 0;
 }
 
-int JvsIo::Jvs_Command_GetCapabilities()
+int JvsIo::Jvs_Command_14_GetCapabilities()
 {
-	ResponseBuffer.push_back(0x01);
+	ResponseBuffer.push_back(ReportCode::Handled);
 
-	// 2 players, 13 switches per player
-	ResponseBuffer.push_back(0x01);
-	ResponseBuffer.push_back(2);
-	ResponseBuffer.push_back(13);
-	ResponseBuffer.push_back(0);
-
-	// Two Coin Slots
-	ResponseBuffer.push_back(0x02);
-	ResponseBuffer.push_back(2);
-	ResponseBuffer.push_back(0);
-	ResponseBuffer.push_back(0);
-
-	// Analog Input, 8 channels, 16 bits per channel
-	ResponseBuffer.push_back(0x03);
-	ResponseBuffer.push_back(8);
-	ResponseBuffer.push_back(16);
-	ResponseBuffer.push_back(0);
-
-	// 6 Outputs
-	ResponseBuffer.push_back(0x12);
-	ResponseBuffer.push_back(6);
-	ResponseBuffer.push_back(0);
-	ResponseBuffer.push_back(0);
+	// Capabilities list (4 bytes each)
+
+	{ // Input capabilities
+		ResponseBuffer.push_back(0x01); // Player button-sets
+		ResponseBuffer.push_back(JVS_MAX_PLAYERS); // number of players
+		ResponseBuffer.push_back(13); // 13 button switches per player
+		ResponseBuffer.push_back(0);
+
+		ResponseBuffer.push_back(0x02); // Coin slots
+		ResponseBuffer.push_back(JVS_MAX_COINS); // number of coin slots
+		ResponseBuffer.push_back(0);
+		ResponseBuffer.push_back(0);
+
+		ResponseBuffer.push_back(0x03); // Analog inputs
+		ResponseBuffer.push_back(JVS_MAX_ANALOG); // number of analog input channels
+		ResponseBuffer.push_back(16); // 16 bits per analog input channel
+		ResponseBuffer.push_back(0);
+
+		// TODO : Rotary input (0x04), JVS_MAX_ROTARY, 0, 0

```

Comment:
There is nothing TODO here: Our target JVS I/O board does not support these additional capabilities.

Reply:
I'll remove those, then.